tests/sentry/hybrid_cloud/__init__.py

```diff
@@ -47,11 +47,13 @@ def validate_relation_does_not_cross_silo_foreign_keys(
     model: Type[Model],
     related: Type[Model],
 ):
-    for mode in model._meta.silo_limit.modes:
-        if mode not in related._meta.silo_limit.modes:
-            raise ValueError(
-                f"{model!r} runs in {mode}, but is related to {related!r} which does not.  Add this relationship pair as an exception or drop the foreign key."
-            )
+    model_modes = model._meta.silo_limit.modes
+    related_modes = related._meta.silo_limit.modes
+    overlap = model_modes.intersection(related_modes)
+    if not overlap:
```

Comment:
This appears to not work because `not frozenset({}) == True`. Intuitively, this is broken because Actors and Users are indeed in different silos and so we should not be able to follow foreign keys.

Reply:
yeah, so this change is definitely wrong.  It is not obvious or intuitive, but every `SiloLimit` contains `Monolith` mode, thus there will always be overlap and it would allow every relation.  The original implementation is more akin to if overlap != original (that is, every element in a belongs in b).  The loop was merely to make the error message more helpful by calling out which silo mode, specifically, is the problem.

Reply:
The idea for this PR is that Actor can both be in control and region silos without being replicated. So that basis is faulty to begin with.

Reply:
Thanks for clarifying that each `SiloLimit` contains `Monolith` mode. I'll work with that assumption.
Is the `Monolith` expected to be available via `SiloLimit.modes`. Doesn't seem to be the case currently.

```
>>> from sentry.db.models import control_silo_only_model, region_silo_only_model
>>> control_silo_only_model.modes
frozenset({<SiloMode.CONTROL: 'CONTROL'>})
>>> region_silo_only_model.modes
frozenset({<SiloMode.REGION: 'REGION'>})
```